datafusion/core/src/datasource/listing/table.rs

```diff
@@ -114,19 +114,22 @@ impl ListingTableConfig {
         }
     }
 
-    fn infer_file_extension(path: &str) -> Result<String> {
+    fn infer_file_extension_and_compression_type(
```

Comment:
Can you please document what the return values are? Something like

```suggestion
    /// Returns a tuple of (file_extension, optional compression_extension)
    ///
    /// For example `("csv", Some("gz"))`
    fn infer_file_extension_and_compression_type(
```

Reply:
Added the documentation as suggested